components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/JdbcTemplate.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.database.utils.jdbc;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.MessageFormatter;
+import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+
+/**
+ * A utility class to call JDBC with lambda expressions. This class simplifies the usage of JDBC and helps to avoid
+ * common errors.
+ * Using this class we can execute insert, select, update and delete queries by providing necessary parameters to the
+ * methods. Transaction support is also available with this implementation for the query executions. All the SQL
+ * operations performed by this class are logged at the debug level.
+ */
+public class JdbcTemplate {
+
+    private static final Logger log = LoggerFactory.getLogger(JdbcTemplate.class);
+    private static final String ID = "ID";
+    private static String driverName;
+    private DataSource dataSource;
+
+    public JdbcTemplate(DataSource dataSource) {
+
+        this.dataSource = dataSource;
+    }
+
+    /**
+     * Provides the transaction support for the JDBC executions of the template.
+     *
+     * @param callable the SQL query execution method call.
```

Comment:
Shall we start the sentences with an uppercase letter?
Shall we fix this in other places?